worlds/overcooked2/Logic.py

```diff
@@ -35,17 +35,13 @@ def has_requirements_for_level_star(
         state: CollectionState, level: Overcooked2GenericLevel, stars: int, player: int) -> bool:
     assert 0 <= stars <= 3
 
-    # First ensure that previous stars are obtainable
-    if stars > 1:
-        if not has_requirements_for_level_star(state, level, stars-1, player):
-            return False
-
-    # Second, ensure that global requirements are met
+    # First, ensure that global requirements for this many stars are met.
+    # Lower numbers of stars are implied meetable if this level is meetable.
     if not meets_requirements(state, "*", stars, player):
         return False
 
-    # Finally, return success only if this level's requirements are met
-    return meets_requirements(state, level.shortname, stars, player)
+    # Then return success only if this level's requirements are met at all stars up through this one
+    return all(meets_requirements(state, level.shortname, s, player) for s in range(1, stars + 1))
```

Comment:
Maybe this is some weird OC2 logic thing, but why does this check for multiple values that exclusively get larger? Can you not just check for `stars`?

Reply:
Yeah, that's OC2 logic. Previously if stars=3 for example, it would call itself with stars=2 and stars=1 to validate access to those stars. (Can't get the third star if you can't get the first, after all.) A future optimization would be to merge those level requirements when possible but for now it does have to do all of them.

As for doing it in ascending order, that's also preserving prior behavior (stars=3 depends on stars=2 depends on stars=1, so the lowest `meets_requirements` is called first).

Reply:
FWIW, reversing the order of the calls doesn't change item placements, though the speed increase is extremely minimal

Reply:
That's about what I expected.